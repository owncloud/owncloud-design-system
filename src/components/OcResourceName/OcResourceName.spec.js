import { shallowMount } from "@vue/test-utils"

import Name from "./OcResourceName.vue"

const fullPath = "images/nature/forest.jpg"
const name = "forest.jpg"
const extension = "jpg"
const type = "file"

describe("OcResourceName", () => {
  it("doesn't add extension to hidden folder", () => {
    const wrapper = shallowMount(Name, {
      propsData: {
        fullPath: ".hidden",
        name: ".hidden",
        extension: "",
        type: "folder",
      },
    })

    expect(wrapper.find(".oc-resource-basename").text()).toMatch(".hidden")
    expect(wrapper.find(".oc-resource-extension").exists()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it("renders folder names with dots completely in the basename", () => {
    const wrapper = shallowMount(Name, {
      propsData: {
        fullPath: "folder.with.dots",
        name: "folder.with.dots",
        extension: "",
        type: "folder",
      },
    })

    expect(wrapper.find(".oc-resource-basename").text()).toMatch("folder.with.dots")
    expect(wrapper.find(".oc-resource-extension").exists()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it("has properties for resource path, name and type", () => {
    const wrapper = shallowMount(Name, {
      propsData: {
        fullPath,
        name,
        extension,
        type,
      },
    })

    const node = wrapper.find(".oc-resource-name")
    expect(node.attributes("data-test-resource-type")).toMatch(type)
    expect(node.attributes("data-test-resource-name")).toMatch(name)
    expect(node.attributes("data-test-resource-path")).toMatch("/" + name)
  })

  it("truncates very long name per default", () => {
    const wrapper = shallowMount(Name, {
      propsData: {
        fullPath:
          "super-long-file-name-which-will-be-truncated-when-exceeding-the-screen-space-while-still-preserving-the-file-extension-at-the-end.txt",
        name: ".super-long-file-name-which-will-be-truncated-when-exceeding-the-screen-space-while-still-preserving-the-file-extension-at-the-end.txt",
        extension: "txt",
        type: "file",
      },
    })

    expect(wrapper).toMatchSnapshot()
  })

  it("does not truncate a very long name when disabled", () => {
    const wrapper = shallowMount(Name, {
      propsData: {
        fullPath:
          "super-long-file-name-which-will-be-truncated-when-exceeding-the-screen-space-while-still-preserving-the-file-extension-at-the-end.txt",
        name: ".super-long-file-name-which-will-be-truncated-when-exceeding-the-screen-space-while-still-preserving-the-file-extension-at-the-end.txt",
        extension: "txt",
        type: "file",
        truncateName: false,
      },
    })

    expect(wrapper).toMatchSnapshot()
  })

  it("shows the name as HTML title if the path is not displayed", () => {
    const wrapper = shallowMount(Name, {
      propsData: {
        fullPath: "folder",
        name: "folder",
        extension: "",
        type: "folder",
        isPathDisplayed: false,
      },
    })

    expect(wrapper).toMatchSnapshot()
  })

  it("does not show the name as HTML title if the path is being displayed", () => {
    const wrapper = shallowMount(Name, {
      propsData: {
        fullPath: "folder",
        name: "folder",
        extension: "",
        type: "folder",
        isPathDisplayed: true,
      },
    })

    expect(wrapper).toMatchSnapshot()
  })
})
