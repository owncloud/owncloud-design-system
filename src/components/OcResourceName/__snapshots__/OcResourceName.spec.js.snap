// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcResourceName does not show the name as HTML title if the path is being displayed 1`] = `
<span data-test-resource-path="folder" data-test-resource-name="folder" data-test-resource-type="folder" class="oc-resource-name"><span class="oc-text-truncate"><span class="oc-resource-basename">folder</span></span>
<!----></span>
`;

exports[`OcResourceName does not truncate a very long name when disabled 1`] = `<span data-test-resource-path="super-long-file-name-which-will-be-truncated-when-exceeding-the-screen-space-while-still-preserving-the-file-extension-at-the-end.txt" data-test-resource-name=".super-long-file-name-which-will-be-truncated-when-exceeding-the-screen-space-while-still-preserving-the-file-extension-at-the-end.txt" data-test-resource-type="file" title=".super-long-file-name-which-will-be-truncated-when-exceeding-the-screen-space-while-still-preserving-the-file-extension-at-the-end.txt" class="oc-resource-name oc-display-inline-block"><span class="oc-resource-basename oc-text-break">.super-long-file-name-which-will-be-truncated-when-exceeding-the-screen-space-while-still-preserving-the-file-extension-at-the-end</span><span class="oc-resource-extension">.txt</span></span>`;

exports[`OcResourceName doesn't add extension to hidden folder 1`] = `
<span data-test-resource-path=".hidden" data-test-resource-name=".hidden" data-test-resource-type="folder" title=".hidden" class="oc-resource-name"><span class="oc-text-truncate"><span class="oc-resource-basename">.hidden</span></span>
<!----></span>
`;

exports[`OcResourceName renders folder names with dots completely in the basename 1`] = `
<span data-test-resource-path="folder.with.dots" data-test-resource-name="folder.with.dots" data-test-resource-type="folder" title="folder.with.dots" class="oc-resource-name"><span class="oc-text-truncate"><span class="oc-resource-basename">folder.with.dots</span></span>
<!----></span>
`;

exports[`OcResourceName shows the name as HTML title if the path is not displayed 1`] = `
<span data-test-resource-path="folder" data-test-resource-name="folder" data-test-resource-type="folder" title="folder" class="oc-resource-name"><span class="oc-text-truncate"><span class="oc-resource-basename">folder</span></span>
<!----></span>
`;

exports[`OcResourceName truncates very long name per default 1`] = `<span data-test-resource-path="super-long-file-name-which-will-be-truncated-when-exceeding-the-screen-space-while-still-preserving-the-file-extension-at-the-end.txt" data-test-resource-name=".super-long-file-name-which-will-be-truncated-when-exceeding-the-screen-space-while-still-preserving-the-file-extension-at-the-end.txt" data-test-resource-type="file" title=".super-long-file-name-which-will-be-truncated-when-exceeding-the-screen-space-while-still-preserving-the-file-extension-at-the-end.txt" class="oc-resource-name"><span class="oc-text-truncate"><span class="oc-resource-basename">.super-long-file-name-which-will-be-truncated-when-exceeding-the-screen-space-while-still-preserving-the-file-extension-at-the-end</span></span><span class="oc-resource-extension">.txt</span></span>`;
